zebra-chain/src/test/vectors.rs

```diff
@@ -0,0 +1,157 @@
+//! Network methods for fetching blockchain vectors.
+//!
+
+use std::collections::BTreeMap;
+
+use crate::{block::Block, parameters::Network, serialization::ZcashDeserializeInto};
+
+use zebra_test::vectors::{
+    BLOCK_MAINNET_1046400_BYTES, BLOCK_MAINNET_653599_BYTES, BLOCK_MAINNET_982681_BYTES,
+    BLOCK_TESTNET_1116000_BYTES, BLOCK_TESTNET_583999_BYTES, BLOCK_TESTNET_925483_BYTES,
+    CONTINUOUS_MAINNET_BLOCKS, CONTINUOUS_TESTNET_BLOCKS, MAINNET_BLOCKS,
+    MAINNET_FINAL_SAPLING_ROOTS, MAINNET_FINAL_SPROUT_ROOTS,
+    SAPLING_FINAL_ROOT_MAINNET_1046400_BYTES, SAPLING_FINAL_ROOT_TESTNET_1116000_BYTES,
+    TESTNET_BLOCKS, TESTNET_FINAL_SAPLING_ROOTS, TESTNET_FINAL_SPROUT_ROOTS,
+};
+
+/// Network methods for fetching blockchain vectors.
+impl Network {
+    /// Returns true if network is of type Mainnet.
+    pub fn is_mainnet(&self) -> bool {
+        matches!(self, Network::Mainnet)
+    }
+
+    /// Returns iterator over blocks.
+    pub fn get_block_iter(&self) -> std::collections::btree_map::Iter<'static, u32, &'static [u8]> {
+        if self.is_mainnet() {
+            MAINNET_BLOCKS.iter()
+        } else {
+            TESTNET_BLOCKS.iter()
+        }
+    }
+
+    ///
+    pub fn get_block_map(&self) -> &BTreeMap<u32, &'static [u8]> {
+        if self.is_mainnet() {
+            &*zebra_test::vectors::MAINNET_BLOCKS
+        } else {
+            &*zebra_test::vectors::TESTNET_BLOCKS
+        }
+    }
+
+    /// Returns genesis block for chain.
+    pub fn get_gen_block(&self) -> std::option::Option<&&[u8]> {
+        if self.is_mainnet() {
+            MAINNET_BLOCKS.get(&0)
+        } else {
+            TESTNET_BLOCKS.get(&0)
+        }
+    }
+
+    /// Returns block bytes
+    pub fn get_test_block(&self, main_height: u32, test_height: u32) -> Option<Block> {
+        let block_bytes = match (self.is_mainnet(), main_height, test_height) {
```

Comment:
There's a clippy warning for returning this match statement directly. There'll likely be one for the `get_test_block_sapling_roots()` method below as well.

Optional/Nitpick: We conventionally omit the `get_` prefix from getter methods in Rust.

Reply:
fixed all clippy warnings and removed get_ prefix from getter methods.